src/coreclr/src/gc/unix/configure.cmake

```diff
@@ -72,17 +72,15 @@ check_cxx_source_runs("
 check_library_exists(pthread pthread_condattr_setclock "" HAVE_PTHREAD_CONDATTR_SETCLOCK)
 
 check_cxx_source_runs("
-    #include <stdlib.h>
-    #include <mach/mach_time.h>
-    int main()
-    {
-        int ret;
-        mach_timebase_info_data_t timebaseInfo;
-        ret = mach_timebase_info(&timebaseInfo);
-        mach_absolute_time();
-        exit(ret);
-    }
-    " HAVE_MACH_ABSOLUTE_TIME)
+#include <stdlib.h>
+#include <time.h>
+
+int main()
+{
+  int ret;
+  ret = clock_gettime_nsec_np(CLOCK_UPTIME_RAW);
```

Comment:
Just curious - what was wrong with `mach_absolute_time` ?

Reply:
It doesn't exist on aarch64 and Apple doc for mach_absolute_time recommends to replace it with clock_gettime_nsec_np

Reply:
See https://developer.apple.com/documentation/kernel/1462446-mach_absolute_time

Reply:
I have noticed that `HAVE_MACH_ABSOLUTE_TIME` is still used under libraries. Do we need to fix it there as well?

Reply:
Hmm, the strange thing is that libraries compile for arm64 with it just fine. So maybe the failure due to which I've modified the usage of mach_absolute_time to clock_gettime_nsec_np was something temporary in the middle of the fixes I was making. Anyways, the change is good since the mach_absolute_time is deprecated, so we should change it in the libraries too.

Reply:
Ok, opened https://github.com/dotnet/runtime/issues/43320